example_clients/pmacmmirs.py

```diff
@@ -0,0 +1,345 @@
+                    float(detenttolb[0] ),
+
+                    float(detenttol[0] ),
+                    
+                    ))
+                break
+            break
+            
+            await pmac.run(wheel, target+5, "Abs")
+
+            for pos in reversed(testpos):
+                pos = pos%360
+                await pmac.run(wheel, pos, "Abs")
+                print(f"moved {wheel} to test position {pos}")
+                p = await pmac.get(f"{wheel}Pos")
+                h = await pmac.get(f"{wheel}HLimBit")
+                m = await pmac.get(f"{wheel}MLimBit")
+                awaylima = await pmac.get(f"{wheel}DetentAwayLimA")
+                awaylimb = await pmac.get(f"{wheel}DetentAwayLimB")
+                awaylima = await pmac.get(f"{wheel}DetentAwayLimA")
+                awaylimb = await pmac.get(f"{wheel}DetentAwayLimB")
+
+                nearlima = await pmac.get(f"{wheel}DetentNearLimA")
+                nearlimb = await pmac.get(f"{wheel}DetentNearLimB")
+                nearlima = await pmac.get(f"{wheel}DetentNearLimA")
+                nearlimb = await pmac.get(f"{wheel}DetentNearLimB")
+
+                swhistaway = await pmac.get(f"{wheel}SwitchHistAway")
+                swhistnear = await pmac.get(f"{wheel}SwitchHistNear")
+
+                detenttola = await pmac.get(f"{wheel}DetentTola")
+                detenttolb = await pmac.get(f"{wheel}DetentTolB")
+
+                detenttol = await pmac.get(f"{wheel}DetentTol")
+                now = time.time()-start
+
+                now = time.time()-start
+
+                data.append((
+                    now, 
+                    float(p[0]), 
+                    int(h[0]), 
+                    int(m[0]), 
+                    wheel, 
+                    "positive", 
+                    float(awaylima[0] ),
+                    float(awaylimb[0] ),
+                    float(awaylima[0] ),
+                    float(awaylimb[0] ),
+
+                    float(nearlima[0] ),
+                    float(nearlimb[0] ),
+                    float(nearlima[0] ),
+                    float(nearlimb[0] ),
+
+                    float(swhistaway[0] ),
+                    float(swhistnear[0] ),
+
+                    float(detenttola[0] ),
+                    float(detenttolb[0] ),
+
+                    float(detenttol[0] ),
+                    
+                    ))
+
+
+        break
+    df = pd.DataFrame(data, 
+        columns=("time", "pos", "h", 'm', 'wheel',
+        'direction',
+
+                "awaylima",
+                "awaylimb",
+                "awaylima",
+                "awaylimb",
+
+                "nearlima",
+                "nearlimb",
+                "nearlima",
+                "nearlimb",
+
+                "swhistaway",
+                "swhistnear",
+
+                "detenttola",
+                "detenttolb",
+
+                "detenttol",
+
+        ))
+    
+#    try:
```

Comment:
if this commented code was for testing, should probably clear it out before merging.